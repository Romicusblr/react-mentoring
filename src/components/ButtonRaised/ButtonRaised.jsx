/* eslint-disable react/button-has-type */
import React from 'react';
import classNames from 'classnames';
import PropTypes from 'prop-types';
import style from './ButtonRaised.module.css';

function Button({
  type, children, className, ...other
}) {
  return (
    <button
      type={type}
      className={classNames(style.buttonRaised, className)}
      {...other}
    >
      {children}
    </button>
  );
}

Button.propTypes = {
  children: PropTypes.node.isRequired,
  type: PropTypes.string,
  className: PropTypes.string,
};

Button.defaultProps = {
  type: 'submit',
  className: '',
};

export default Button;
