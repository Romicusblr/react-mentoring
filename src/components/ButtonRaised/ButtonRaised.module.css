.button {
  width: 235px;
  padding: 1rem;
  margin-left: .5rem;
  background-color: var(--color-raised);
  color: var(--color-white);
  text-transform: uppercase;
  border-radius: .25rem;
  border-color: transparent
}
