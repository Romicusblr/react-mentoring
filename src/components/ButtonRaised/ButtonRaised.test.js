import ButtonRaised from '.';

describe('ButtonRaised', () => {
  it('should match to snapshot', () => {
    const snap = shallow(<ButtonRaised>text</ButtonRaised>);
    expect(snap).toMatchSnapshot();
  });
});
