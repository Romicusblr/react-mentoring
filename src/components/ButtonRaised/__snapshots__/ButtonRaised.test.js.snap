// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonRaised should match to snapshot 1`] = `
<button
  className="buttonRaised"
  type="submit"
>
  text
</button>
`;
